### PR TITLE
[image_picker] add requestFullMetadata for iOS (optional permissions)

### PR DESCRIPTION
**Pull request by PiotrMitkowski**
## Description
Originally #3264 made by @cpboyd 
Creating a new PR as I've closed the previous one (#4638) by a mistake

Make iOS 11+ permissions requests optional per https://developer.apple.com/forums/thread/653414.
PHAssets aren't actually required and there was already some fallback code in place.

## Related issues
Fixes flutter/flutter#65995

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [relevant style guides] and ran [the auto-formatter]. (Unlike the flutter/flutter repo, the flutter/plugins repo does use `dart format`.)
- [x] I signed the [CLA].
- [x] The title of the PR starts with the name of the plugin surrounded by square brackets, e.g. `[shared_preferences]`
- [x] I listed at least one issue that this PR fixes in the description above.
- [x] I updated `pubspec.yaml` with an appropriate new version according to the [pub versioning philosophy], or this PR is [exempt from version changes].
- [x] I updated `CHANGELOG.md` to add a description of the change, [following repository CHANGELOG style].
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] I added new tests to check the change I am making, or this PR is [test-exempt].
- [x] All existing and new tests are passing.

[Contributor Guide]: https://github.com/flutter/plugins/blob/master/CONTRIBUTING.md
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[relevant style guides]: https://github.com/flutter/plugins/blob/master/CONTRIBUTING.md#style
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat
[pub versioning philosophy]: https://dart.dev/tools/pub/versioning
[exempt from version changes]: https://github.com/flutter/flutter/wiki/Contributing-to-Plugins-and-Packages#version-and-changelog-updates
[following repository CHANGELOG style]: https://github.com/flutter/flutter/wiki/Contributing-to-Plugins-and-Packages#changelog-style
[the auto-formatter]: https://github.com/flutter/plugins/blob/master/script/tool/README.md#format-code
[test-exempt]: https://github.com/flutter/flutter/wiki/Tree-hygiene#tests

### CODE REVIEW

**Comment by Hixie** [comment]
@PiotrMitkowski Will you be able to continue working on this? It's totally fine if not; we can take it over (though it might take longer). I'm just asking because I'm trying to clean out our review queue.

**Comment by PiotrMitkowski** [comment]
@Hixie definitely yes, it's just blocked by another PR for iOS changes: https://github.com/flutter/plugins/pull/5713. I'll get back here as soon as those iOS changes are reviewed and merged.

**Comment by stuartmorgan** [comment]
Sorry about that review delay; the iOS changes have now been published!

**Comment by PiotrMitkowski** [comment]
> Sorry about that review delay; the iOS changes have now been published!

No problem, thanks!

**Review by stuartmorgan — COMMENTED** [review]
This looks great! Just one thing I overlooked in the original review: the README section about iOS permissions needs to be updated to discuss optionality of the library permission.

File: packages/image_picker/image_picker/test/image_picker_test.dart
Comment (stuartmorgan) [inline packages/image_picker/image_picker/test/image_picker_test.dart:249]: Minor nit here and below: our usual convention (which the other tests around this follow) is that they start with a lower-case letter.

**Comment by PiotrMitkowski** [comment]
> This looks great! Just one thing I overlooked in the original review: the README section about iOS permissions needs to be updated to discuss optionality of the library permission.

@stuartmorgan I've added a paragraph about new permissions handling. Let me know if that's what you had in mind.

**Review by stuartmorgan — CHANGES_REQUESTED** [review]

**Comment by PiotrMitkowski** [comment]
@idish thanks for your feedback! I'll take an extra look to make sure, that it works correctly.

**Comment by PiotrMitkowski** [comment]
@idish I've found a bug and fixed it in #6429. You can take a look there, if you want. I'll put this PR in draft while waiting on the other one to be merged.

**Comment by idish** [comment]
> @idish I've found a bug and fixed it in #6429. You can take a look there, if you want. I'll put this PR in draft while waiting on the other one to be merged.

Thanks! I can confirm that your bugfix PR works.

File: packages/image_picker/image_picker/pubspec.yaml
Comment (stuartmorgan) [inline packages/image_picker/image_picker/pubspec.yaml:27]: This should be `0.8.6+1` now to pick up the last fix.

**Review by stuartmorgan — CHANGES_REQUESTED** [review]
Looks great! It just needs the final version update for your bugfix and it should be good to go! Thanks again for all the work on this.

@cyanglaz Can you do the secondary sign-off?

File: packages/image_picker/image_picker/pubspec.yaml
Comment (PiotrMitkowski) [reply packages/image_picker/image_picker/pubspec.yaml:27]: I'm still waiting for this version to be published on pub.dev. It seems that the release action failed.

Comment (stuartmorgan) [reply packages/image_picker/image_picker/pubspec.yaml:27]: Oops, sorry about that; I forgot to check that publishing had completed. It looks like there was a flake in the analysis run during post-submit tests, and it's already being re-run.

Comment (stuartmorgan) [reply packages/image_picker/image_picker/pubspec.yaml:27]: Now it's live. Sorry for the delay!

Comment (PiotrMitkowski) [reply packages/image_picker/image_picker/pubspec.yaml]: Should be good to go now, thanks!

**Review by stuartmorgan — APPROVED** [review]

**Comment by stuartmorgan** [comment]
@cyanglaz Ping from triage on this review.

**Comment by liumengchun** [comment]
Hello, have you fixed the problem of 'Select More Photos'? I have also reproduced the problem here

**Comment by PiotrMitkowski** [comment]
@liumengchun this is more like a workaround to not use the gallery permissions at all. They are needed only to access some specific image metadata, which isn't needed in most of the cases.

**Comment by furaiev** [comment]
@PiotrMitkowski thank you, great work 👏

**Pull request merged**
